src/daemon/revaultd.rs

```diff
@@ -345,6 +387,35 @@ impl RevaultD {
         let noise_secret_file = [data_dir_str, "noise_secret"].iter().collect();
         let noise_secret = read_or_create_noise_key(noise_secret_file)?;
 
+        let cpfp_key = if our_man_xpub.is_some() {
+            let cpfp_key_file = [data_dir_str, "cpfp_secret"].iter().collect();
+            let key = read_cpfp_key(cpfp_key_file)?;
+            if let Some(key) = key {
+                // Checking if the key is in the cpfp descriptor
+                let secp_ctx = secp256k1::Secp256k1::signing_only();
+                let pubkey = ExtendedPubKey::from_private(&secp_ctx, &key);
+                cpfp_descriptor
+                    .xpubs()
+                    .iter()
+                    .find(|k| {
+                        if let DescriptorPublicKey::XPub(k) = k {
+                            k.xkey == pubkey
+                        } else {
+                            // It shouldn't happen, but I'm not going to complain...
```

Comment:
I don't understand this comment

Reply:
I'm still unsure what I should do here.
When I wrote the code, I thought "I shouldn't find a non-xpub in a cpfp descriptor. If I do, I'll just ignore it". Now I'm thinking I should just panic:
```
                    .find(|k| {
                        if let DescriptorPublicKey::XPub(k) = k {
                            k.xkey == pubkey
                        } else {
                            // CPFP descriptor always contains xpubs
                            unreachable!();
                        }
                    })
```
What do you think?